### ISSUE TITLE
Gradient explosion while training

### ISSUE THREAD

**Issue by zzz1515151** [XingangPan/SCNN#82]
Hi,
i have rewritten the SCNN codes using pytorch but facing gradient explosion problems (the instance_seg_loss is too big and became Nan at the beginning of the training process).  The program is still running but i wonder is it normal to get Nan loss in SCNN training? Thanks in advance!

**Comment by XingangPan** [comment]
@zzz1515151 There are several things you can try to avoid the divergence problem. 
1. Using smaller initialization value for the SCNN layers. I initialize SCNN layers with variance sqrt(5) times smaller than the MSRA initialization.
2. It would be helpful to initialize all other layers with a pretrained model. Directly training from random initialization may cause SCNN to diverge.
3. You may try using a warm-up learning rate schedule, i.e., using very small learning rate at the beginning and gradually increase the learning rate.

**Comment by zzz1515151** [comment]
wow, these are quite generous advices! i have used ImageNet pretrained
weights and initialiaze the SCNN layers as the tensorflow code did. i'll
try adjusting learning rate to see if it works. Really appreciate your
reply!
Xingang Pan <notifications@github.com> 于 2019年5月31日周五 19:18写道：
